timesketch/frontend-ng/src/components/Visualization/ChartCard.vue

```diff
@@ -0,0 +1,186 @@
+<!--
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ -->
+ <template>
+  <v-card
+    flat
+    :min-height="height"
+    :width="width + (chartType === 'table' ? 25 : 0)"
+  >
+    <v-card-text>
+      <TsBarChart 
+        v-if="chartSeries != null && chartType === 'bar'"
+        :fieldName="fieldName"
+        :metricName="metricName"
+        :is-time-series="isTimeSeries"
+        :chartSeries="chartSeries" 
+        :chartLabels="chartLabels"
+        :chartTitle="chartTitle"
+        :height="height"
+        :width="width"
+        :xTitle="xTitle"
+        :showXLabels="showXLabels"
+        :yTitle="yTitle"
+        :showYLabels="showYLabels"
+        :showDataLabels="showDataLabels"
+      >
+      </TsBarChart>
+      <TsColumnChart
+        v-if="chartSeries != null && chartType === 'column'"
+        :fieldName="fieldName"
+        :metricName="metricName"
+        :is-time-series="isTimeSeries"
+        :chartSeries="chartSeries" 
+        :chartLabels="chartLabels"
+        :chartTitle="chartTitle"
+        :height="height"
+        :width="width"
+        :xTitle="xTitle"
+        :showXLabels="showXLabels"
+        :yTitle="yTitle"
+        :showYLabels="showYLabels"
+        :showDataLabels="showDataLabels"
+      >
+      </TsColumnChart>
+      <TsDonutChart
+        v-if="chartSeries != null && chartType === 'donut'"
+        :fieldName="fieldName"
+        :metricName="metricName"
+        :is-time-series="isTimeSeries"
+        :chartSeries="chartSeries" 
+        :chartLabels="chartLabels"
+        :chartTitle="chartTitle"
+        :height="height"
+        :width="width"
+        :xTitle="xTitle"
+        :showXLabels="showXLabels"
+        :yTitle="yTitle"
+        :showYLabels="showYLabels"
+        :showDataLabels="showDataLabels"
+      >
+      </TsDonutChart>
+      <TsHeatmapChart
+        v-if="chartSeries != null && chartType === 'heatmap'"
+        :fieldName="fieldName"
+        :metricName="metricName"
+        :is-time-series="isTimeSeries"
+        :chartSeries="chartSeries" 
+        :chartLabels="chartLabels"
+        :chartTitle="chartTitle"
+        :height="height"
+        :width="width"
+        :xTitle="xTitle"
+        :showXLabels="showXLabels"
+        :yTitle="yTitle"
+        :showYLabels="showYLabels"
+        :showDataLabels="showDataLabels"
+      >
+      </TsHeatmapChart>
+      <TsLineChart 
```

Comment:
The LineChart but also the other charts except the table have some overlay or cutout with their surrounding box. See screenshot below.
![image](https://github.com/google/timesketch/assets/99879757/e5b51bff-4806-40eb-9192-c24c750d796b)

Reply:
Seems this will occur if the configured chart size is wider than the page dimensions.  

What is the desired behaviour you are expecting when this happens?

Reply:
Interesting. My page dimension has plenty of space left, but this still happens. Can this be some default chat size that does not match? 

I would go with fitting the chat into the borders. Clipping seems to be more of an unexpected behaviour. What do you think?

Reply:
Ok, you are probably right, I'm only working off a laptop screen so don't have much horizontal real estate. 

I think with the user-configurable dimension it might seem counter-intuitive to then shrink it since they would expect the dimension setting to match with what is displayed?

Also, I'm not 100% sure I can scale/shrink the chart once its dimensions are set (as its a 3P library) but I'll see what I can come up with.

Reply:
This happens even with custom width settings for me:
![image](https://github.com/google/timesketch/assets/99879757/706e6f41-7f92-4237-8728-8187e528aeb9)

Reply:
Ah thanks for the screenshot, I think I know what's causing that.

Reply:
See also https://github.com/google/timesketch/pull/3040#discussion_r1609849771 before digging deeper into this.